lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants
+  """
+  import Nx.Defn
+  alias __MODULE__
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and
+  sample targets `y`. Depending on number of classes the function chooses
+  either binary or multinomial logistic regression.
+  """
+  defn(fit(x, y, opts \\ []), do: fit_verify(x, y, opts))
+
+  # Function checks validity of the provided data
+
+  deftransformp fit_verify(x, y, opts) do
+    if !is_integer(opts[:num_classes]) or opts[:num_classes] < 1 do
+      raise ArgumentError, "The number of classes must be a positive integer"
+    end
+
+    if Nx.rank(x.shape) != 2 do
+      raise ArgumentError, "Training vector must be two-dimensional (n_samples, n_features)"
+    end
+
+    if (Nx.rank(y.shape) != 1 and !opts[:one_hot]) or (Nx.rank(y.shape) != 2 and opts[:one_hot]) do
+      raise ArgumentError,
+            "Target vector must be one-dimensional (n_samples) or two-dimensional if :one_hot set to true"
+    end
+
+    if is_number(opts[:lr]) and opts[:lr] < 0 do
```

Comment:
```suggestion
    unless is_number(opts[:lr]) and opts[:lr] >= 0 do
```

Reply:
`[lr: :atom]` passes the current check without raising. Same issue with `iterations` below